### PR TITLE
Main server implementation [Task #767965]

### PR DESCRIPTION
**Pull request by drenzoo**
Implement server interface in real code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,33 +11,27 @@ TST_DIR = tests
 BIN_DIR = bin
 DIRS = $(SRC_DIR) $(OBJ_DIR) $(TST_DIR) $(BIN_DIR)
 
-SERV_SRC = $(SRC_DIR)/server.c $(SRC_DIR)/room.c $(SRC_DIR)/worker.c
-CLNT_SRC = $(SRC_DIR)/client.c
-SERV_OBJ = $(SERV_SRC:$(SRC_DIR)/%.c=$(OBJ_DIR)/%.o)
-CLNT_OBJ = $(CLNT_SRC:$(SRC_DIR)/%.c=$(OBJ_DIR)/%.o)
-SERV = server.out
-CLNT = client.out
+SRC = $(wildcard $(SRC_DIR)/*.c)
```
Comment (FNickRU) [inline Makefile]: Return new version of Makefile, not this

File: check(old).sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+LINUX_VER=$(cat /etc/*-release | grep ID_LIKE)
```
Comment (FNickRU) [inline check(old).sh]: Delete this file

File: include/unit.h [hunk 3]
```diff
@@ -17,12 +17,24 @@
 #define ANS_COUNT 4
 struct unit {
     char *quest;
-    unsigned int q_len;
+    //unsigned int q_len;
```
Comment (FNickRU) [inline include/unit.h]: Remove, not comment

File: include/unit.h [hunk 4]
```diff
@@ -17,12 +17,24 @@
...
     char *ans[ANS_COUNT];
-    unsigned int ans_len[ANS_COUNT];
+    //unsigned int ans_len[ANS_COUNT];
```
Comment (FNickRU) [inline include/unit.h]: Same

File: src/client.c [hunk 5]
```diff
@@ -2,7 +2,8 @@
  * Game client main source code.
  */
 
-int main(int argc, char *argv[])
+/*int main(int argc, char *argv[])
```
Comment (FNickRU) [inline src/client.c]: Remove changes from client

File: src/server.c [hunk 6]
```diff
@@ -1,9 +1,287 @@
...
+
 #include "server.h"
+#include "worker.h"
+#include "room.h"
+
+#define PORT 11115
+#define BACKLOG 4
+
+//Заглушка
+void room_fsm(void *msgid)
```
Comment (FNickRU) [inline src/server.c]: Delete

File: src/server.c [hunk 7]
```diff
@@ -1,9 +1,287 @@
...
+{
+    return 0;
+}
+void worker_fsm(void *msgid)
```
Comment (FNickRU) [inline src/server.c]: Delete

File: src/server.c [hunk 8]
```diff
@@ -1,9 +1,287 @@
...
+{
+    return 0;
+}
+
+
+//TODO: Transer its to another place.
```
Comment (FNickRU) [inline src/server.c]: What does it mean?

File: src/server.c [hunk 9]
```diff
@@ -1,9 +1,287 @@
...
+    cfg->socket;
+    if ((cfg->socket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) < 0) {
+        error_handler(ERROR_INIT_SOCKET);
+        free(worker_handlers);
+        free(room_handlers);
+        free(cfg);
+        exit(4);
+    }
+    
+    //TODO: Get right IP adress.
```
Comment (FNickRU) [inline src/server.c]: Remove working commentary

File: src/server.c [hunk 10]
```diff
@@ -1,9 +1,287 @@
...
+        
+        int serverd;
+        if ((serverd = accept(socket, &client_addr, &socket_length)) < 0) {
+            error_handler(ERROR_ACCEPT);
+            finalize(finalize_conf);
+        }
+        /**
+         * Get message from client
+         */
+        if (recv(serverd, request, 250, 0) <= 0) {
```
Comment (FNickRU) [inline src/server.c]: You shouldn't receive anything, this is worker's job

File: src/server.c (hunk 8 above)
Comment (FNickRU) [reply src/server.c]: Remove. Use standart message structure

File: src/server.c [hunk 11]
```diff
@@ -1,9 +1,287 @@
...
+    finalize(finalize_conf);
+    exit(0);
+    return 0;
+}
 
-int main(int argc, char *argv[])
+int finalize(struct server_conf *conf)
 {
+    close(conf->socket);
+
```
Comment (FNickRU) [inline src/server.c]: Where is message queue close?

**Review by FNickRU — CHANGES_REQUESTED** [review]
+travis errors and warnings
+remove all unnecessary files with units

File: src/server.c (hunk 11 above)
Comment (drenzoo) [reply src/server.c]: +

File: src/server.c (hunk 9 above)
Comment (drenzoo) [reply src/server.c]: +

File: src/server.c (hunk 10 above)
Comment (drenzoo) [reply src/server.c]: +

File: src/server.c (hunk 8 above)
Comment (drenzoo) [reply src/server.c]: +

File: src/server.c (hunk 7 above)
Comment (drenzoo) [reply src/server.c]: +

File: src/server.c (hunk 6 above)
Comment (drenzoo) [reply src/server.c]: +

**Review by FNickRU — APPROVED** [review]

**Comment by FNickRU** [comment]
Seems ok

**Pull request merged**